### PR TITLE
ISSD-1425 Set up "Browse by Capability" searched article tags

### PR DESCRIPTION
**Pull request by rtienpal**

### CODE REVIEW

**Comment by rtienpal** [comment]
ci:test:sf

**Comment by rtienpal** [comment]
ci:test:cf

**Comment by rtienpal** [comment]
ci:test:sf

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 1]
```diff
@@ -1,90 +1,99 @@
...
 <script>
 	let href = window.location.href;
-
 	if (href.endsWith("/")) {
 		href = href.substring(0, href.length - 1);
 		window.location.assign(href);
 	}
 </script>
-
+<#assign
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:9]:
```suggestion

<#assign
```

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 2]
```diff
@@ -1,90 +1,99 @@
...
+/>
+<#list taxonomyCategoryBriefs as taxonomyCategoryBrief>
+<#assign taxonomyVocabularyName = taxonomyCategoryBrief.embeddedTaxonomyCategory.parentTaxonomyVocabulary.name />
+	<#if !taxonomyVocabularies?seq_contains(taxonomyVocabularyName)>
+		<#assign taxonomyVocabularies = taxonomyVocabularies + [taxonomyVocabularyName] />
+	</#if>
+</#list>
+<#assign taxonomyCategoriesMap = {} />
+<#list taxonomyCategoryBriefs as taxonomyCategoryBrief>
+<#assign taxonomyVocabularyName = taxonomyCategoryBrief.embeddedTaxonomyCategory.parentTaxonomyVocabulary.name />
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:22]: why do you make the same assign to the same variable 2 times in a row?
please just declare the variable out of the loop

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 3]
```diff
@@ -1,90 +1,99 @@
...
+	taxonomyVocabularies = []
+/>
+<#list taxonomyCategoryBriefs as taxonomyCategoryBrief>
+<#assign taxonomyVocabularyName = taxonomyCategoryBrief.embeddedTaxonomyCategory.parentTaxonomyVocabulary.name />
+	<#if !taxonomyVocabularies?seq_contains(taxonomyVocabularyName)>
+		<#assign taxonomyVocabularies = taxonomyVocabularies + [taxonomyVocabularyName] />
+	</#if>
+</#list>
+<#assign taxonomyCategoriesMap = {} />
+<#list taxonomyCategoryBriefs as taxonomyCategoryBrief>
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:21]: do you really need to make the same loop again? 
can't you just do it with only one loop?

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 4]
```diff
@@ -1,90 +1,99 @@
...
-			<div class="doc-nav-wrapper-inner">
+<div class="row">
+<div class="col-12 col-md-2 doc-nav-wrapper mobile-nav-hide">
+<div class="doc-nav-wrapper-inner">
 				<div class="d-md-none mobile-doc-nav-toggler" id="mobileDocNavToggler">${languageUtil.get(locale, "documentation-menu", "Documentation Menu")}
 					<button
 						aria-label="Expand Documentation Menu" class="btn expand-btn" onclick="javascript:;"
 						title="Expand Documentation Menu" type="button">
-						<@clay["icon"] symbol="angle-down-small" />
+<@clay["icon"] symbol="angle-down-small" />
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:70]: this should be correctly spaced

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 5]
```diff
@@ -1,90 +1,99 @@
...
 					</button>
 
 					<button
 						aria-label="Close Documentation Menu" class="btn collapse-btn" onclick="javascript:;"
 						title="Close Documentation Menu" type="button">
-						<@clay["icon"] symbol="angle-up-small" />
+<@clay["icon"] symbol="angle-up-small" />
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:76]: same here

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 6]
```diff
@@ -1,90 +1,99 @@
...
-					</div>
-
+<div class="doc-nav">
 					<#if !topLevelArticle>
-						<a class="back-link btn btn-link btn-monospaced d-flex flex-row justify-content-start" href="${parentLink.url}" id="backLink">
-							<svg class="lexicon-icon lexicon-icon-angle-left" role="presentation" viewBox="0 0 512 512">
-								<use xlink:href="#angle-left" />
+<a class="back-link btn btn-link btn-monospaced d-flex flex-row justify-content-start" href="${parentLink.url}" id="backLink">
+<svg class="lexicon-icon lexicon-icon-angle-left" role="presentation" viewBox="0 0 512 512">
+<use xlink:href="#angle-left" />
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:83]: please do not delete all /t

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 7]
```diff
@@ -1,90 +1,99 @@
...
 		window.location.assign(href);
 	}
 </script>
-
+<#assign
+	journalArticleId = .vars["reserved-article-id"].data
+	taxonomyCategoryBriefs = restClient.get("/headless-delivery/v1.0/sites/${groupId}/structured-contents/by-key/${journalArticleId}?nestedFields=embeddedTaxonomyCategory").taxonomyCategoryBriefs
+	taxonomyVocabularies = []
+/>
+<#list taxonomyCategoryBriefs as taxonomyCategoryBrief>
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:14]: please add blank line after the end of a tag />

**Review by nNilton — CHANGES_REQUESTED** [review]

**Comment by rtienpal** [comment]
ci:test:sf

**Comment by rtienpal** [comment]
ci:test:sf

**Comment by rtienpal** [comment]
ci:test:sf

**Comment by rtienpal** [comment]
ci:test:sf

**Comment by rtienpal** [comment]
ci:test:sf

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 8]
```diff
@@ -1,35 +1,64 @@
...
 		href = href.substring(0, href.length - 1);
 		window.location.assign(href);
 	}
 </script>
 
+<#assign
+	journalArticleId = .vars["reserved-article-id"].data
+	taxonomyCategoryBriefs = restClient.get("/headless-delivery/v1.0/sites/${groupId}/structured-contents/by-key/${journalArticleId}?nestedFields=embeddedTaxonomyCategory").taxonomyCategoryBriefs
+	taxonomyVocabularies = []
+	taxonomyCategoriesMap = {}
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:14]: please sort

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 9]
```diff
@@ -1,35 +1,64 @@
...
+/>
+
+<#list taxonomyCategoryBriefs as taxonomyCategoryBrief>
+<#assign taxonomyVocabularyName = taxonomyCategoryBrief.embeddedTaxonomyCategory.parentTaxonomyVocabulary.name />
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:18]: tab is missing i guess

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 10]
```diff
@@ -1,35 +1,64 @@
...
+		/>
+	</#if>
+</#list>
+
 <#assign
 	groupFriendlyURL = "/web" + themeDisplay.getScopeGroup().getFriendlyURL()
 	isLandingPage = false
 	topLevelArticle = true
 />
-
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:17]: are there any reasons to remove this blank line?

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 11]
```diff
@@ -1,35 +1,64 @@
...
 <#if (breadcrumbLinks.getData())??>
 	<#assign breadcrumbLinksJSONArray = jsonFactoryUtil.createJSONArray(breadcrumbLinks.getData()) />
-
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:20]: are there any reasons to remove this blank line?

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 12]
```diff
@@ -1,35 +1,64 @@
...
 	<#if breadcrumbLinksJSONArray.length() gt 0>
 		<#assign
 			parentLink = breadcrumbLinksJSONArray.getJSONObject(0)?eval
 			topLevelArticle = false
 		/>
 	</#if>
 </#if>
-
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:28]: are there any reasons to remove this blank line?

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 13]
```diff
@@ -1,35 +1,64 @@
...
 <#if (landingPage.getData())?? && (landingPage.getData() == "true")>
 	<#assign isLandingPage = true />
 </#if>
-
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:32]: are there any reasons to remove this blank line?

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 14]
```diff
@@ -38,31 +67,17 @@
 					<button
 						aria-label="Expand Documentation Menu" class="btn expand-btn" onclick="javascript:;"
 						title="Expand Documentation Menu" type="button">
-						<@clay["icon"] symbol="angle-down-small" />
+							<@clay["icon"] symbol="angle-down-small" />
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:70]: remove 1 tab

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 15]
```diff
@@ -38,31 +67,17 @@
...
 					</button>
 
 					<button
 						aria-label="Close Documentation Menu" class="btn collapse-btn" onclick="javascript:;"
 						title="Close Documentation Menu" type="button">
-						<@clay["icon"] symbol="angle-up-small" />
+							<@clay["icon"] symbol="angle-up-small" />
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:76]: remove 1 tab

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 16]
```diff
@@ -71,9 +86,7 @@
 							${languageUtil.get(locale, "go-back", "Go Back")}
 						</a>
 					</#if>
-
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:74]: are there any reasons to remove this blank line?

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 17]
```diff
@@ -71,9 +86,7 @@
...
 					<#if (navigationLinks.getData())??>
-
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:76]: are there any reasons to remove this blank line?

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 18]
```diff
@@ -82,9 +95,7 @@
 									<a class="reference internal" href="${parentLink.url}">${parentLink.title}</a>
 								</li>
 							</#if>
-
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:85]: are there any reasons to remove this blank line?

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl [hunk 19]
```diff
@@ -82,9 +95,7 @@
...
 							<#assign navigationLinksJSONArray = jsonFactoryUtil.createJSONArray(navigationLinks.getData()) />
-
```
Comment (nNilton) [inline workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:87]: are there any reasons to remove this blank line?

**Review by nNilton — CHANGES_REQUESTED** [review]

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl (hunk 8 above)
Comment (rtienpal) [reply workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:14]: done

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl (hunk 9 above)
Comment (rtienpal) [reply workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:18]: done

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl (hunk 14 above)
Comment (rtienpal) [reply workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:70]: This element is inside button, so I believe should be a tab

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl (hunk 15 above)
Comment (rtienpal) [reply workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:76]: This element is inside button, so I believe there should be a tab

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl (hunk 10 above)
Comment (rtienpal) [reply workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:17]: No, it is now added back

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl (hunk 11 above)
Comment (rtienpal) [reply workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:20]: No, corrected now

File: workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl (hunk 13 above)
Comment (rtienpal) [reply workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl:32]: No, corrected now

**Comment by rtienpal** [comment]
ci:test:sf

**Pull request closed without merging**